src/net/codec.rs

```diff
@@ -749,6 +752,131 @@ impl StacksMessageCodec for MemPoolSyncData {
     }
 }
 
+impl StacksMessageCodec for ContractId {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
+        write_next(fd, &self.0.issuer.0)?;
+        write_next(fd, &self.0.issuer.1)?;
+        write_next(fd, &self.0.name)?;
+        Ok(())
+    }
+
+    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<ContractId, codec_error> {
+        let version: u8 = read_next(fd)?;
+        let bytes: [u8; 20] = read_next(fd)?;
+        let name: ContractName = read_next(fd)?;
+        let qn = QualifiedContractIdentifier::new(StandardPrincipalData(version, bytes), name);
+        Ok(ContractId(qn))
+    }
+}
```

Comment:
When you use an extension trait, `impl StacksMessageCodec for QualifiedContractId` would have to be implemented in the `clarity` crate.

Reply:
Instead, I just add special serialization/deserialization helpers before I saw this comment.  Hope that works!